js/completionCalculations.js

```diff
@@ -5,142 +5,149 @@ define([
+                children.forEach(function(contentObject) {
+                    var completionObject = calculateCompletion(contentObject);
+                    subProgressCompleted += contentObjectModel.subProgressCompleted || 0;
+                    subProgressTotal += contentObjectModel.subProgressTotal || 0;
+                    nonAssessmentComponentsTotal += completionObject.nonAssessmentTotal;
+                    nonAssessmentComponentsCompleted += completionObject.nonAssessmentCompleted;
+                    assessmentComponentsTotal += completionObject.assessmentTotal;
+                    assessmentComponentsCompleted += completionObject.assessmentCompleted;
+                });
+
+                return {
+                    "subProgressCompleted": subProgressCompleted,
+                    "subProgressTotal" : subProgressTotal,
+                    "nonAssessmentCompleted": nonAssessmentComponentsCompleted,
+                    "nonAssessmentTotal": nonAssessmentComponentsTotal,
+                    "assessmentCompleted": assessmentComponentsCompleted,
+                    "assessmentTotal": assessmentComponentsTotal,
+                };
         }
     }
 
     function getNonAssessmentComponents(models) {
-        return _.filter(models, function(model) {
+        return models.filter(function(model) {
             return !model.get('_isPartOfAssessment');
         });
     }
 
     function getAssessmentComponents(models) {
-        return _.filter(models, function(model) {
+        return models.filter(function(model) {
             return model.get('_isPartOfAssessment');
         });
     }
 
     function getComponentsCompleted(models) {
-        return _.filter(models, function(item) {
+        return models.filter(function(item) {
             return item.get('_isComplete');
         });
     }
 
     function getComponentsInteractionCompleted(models) {
-        return _.filter(models, function(item) {
+        return models.filter(function(item) {
             return item.get('_isComplete');
         });
     }
 
     //Get only those models who were enabled for pageLevelProgress
     function getPageLevelProgressEnabledModels(models) {
-        return _.filter(models, function(model) {
-            if (model.get('_pageLevelProgress')) {
-                return model.get('_pageLevelProgress')._isEnabled;
-            }
+        return models.filter(function(model) {
+            var config = model.get('_pageLevelProgress');
+            return config && config._isEnabled;
         });
     }
 
     function unavailableInHierarchy(parents) {
-        if (parents.length > 0) {
-            var parentsAvailable = _.map(parents, function(parent) {
-                return parent.get('_isAvailable');
-            });
-            return _.indexOf(parentsAvailable, false) > -1;
-        } else {
-            return;
-        }
+        if (parents.length === 0) return;
```

Comment:
Amend to
```js
return parents && parents.some(function(parent) {
```
or
```js
if (!parents) return;
```

[`AdaptModel.getAncestorModels()`](https://github.com/adaptlearning/adapt-contrib-pageLevelProgress/pull/107/files#diff-de9ea7c4944218ae10c3ff683fb7f444R129) would return `null` rather than `[]`.